JRC Publications Repository.js

```diff
@@ -0,0 +1,477 @@
+	"Articles in Journals" : "journalArticle",
+	"Articles in books" : "bookSection",
+	"Books" : "book",
+	"eBook" : "book",
+	"Contributions to Conferences" : "conferencePaper",
+	"JRC Reference Reports" : "report",
+	"EUR - Scientific and Technical Research Reports" : "report",
+	"PhD Theses" : "thesis"
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//div[contains(@class, "discovery-result-results") or contains(@class, "panel-primary")]//table//a[contains(@href, "/repository/handle/")]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function detectWeb(doc, url) {
+	var type = ZU.xpath(doc, '//meta[contains(@name, "DC.type")]/@content');
+	if (type.length>0) {
+		if (mappingTable[type[0].textContent]) {
+			return mappingTable[type[0].textContent];
+		} else {//generic fallback
+			Z.debug('Unrecognized item type: ' + type[0].textContent);
+			return "journalArticle";
+		}
+	}
+	if ( getSearchResults(doc, true) ) {
+		return "multiple";
+	}
+}
+
+//We need a different name for the same function, because of calling another translator inside scrape makes problems.
+function detectWebHere(doc, url) {
+	var type = ZU.xpath(doc, '//meta[contains(@name, "DC.type")]/@content');
+	if (type.length>0) {
+		if (mappingTable[type[0].textContent]) {
+			return mappingTable[type[0].textContent];
+		} else {//generic fallback
+			Z.debug('Unrecognized item type: ' + type[0].textContent);
+			return "journalArticle";
+		}
+	}
+	if ( getSearchResults(doc, true) ) {
+		return "multiple";
+	}
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = new Array();
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, url) {
+	//Calling detectWeb here on multiples, does only work for the first one?!
+	Z.debug( 'DETECTWEB = ' + detectWeb(doc, url) );//I guess this then tries (for the nth (n>1) entry) to use the detectWeb function form the called Metadata translator.
+	Z.debug( 'DETECTWEBHERE = ' + detectWebHere(doc, url) );//This always uses the correct function.
+	var itemType = detectWebHere(doc, url);
+	
+	var trans = Zotero.loadTranslator('web');
+	trans.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');//https://github.com/zotero/translators/blob/master/Embedded%20Metadata.js
+	trans.setDocument(doc);
+	
+	trans.setHandler('itemDone', function (obj, item) {
+		
+		item.itemType = itemType;
+		
+		if (item.title == item.title.toUpperCase()) {
+			item.title = ZU.capitalizeTitle( item.title, true );
```

Comment:
I'm not so sure about publication titles. Generally, sentence case is better, but if most styles assume that publication titles to already be in title case, then no. CC @adam3smith